### PR TITLE
[CR]Make explosion distance factor a parameter

### PR DESCRIPTION
**Pull request by Coolthulhu**
Currently the explosions are hardcoded to lose roughly 20% of their power per tile traveled.

Here I moved this 20% to an argument. This allows creating concentrated explosions and huge expanding areas of mostly low damage.
Explosion intensity at a tile is: `base_intensity * distance_factor^distance` (distance is roughly 10% larger than actual distance, due to random air resistance factor added for less predictability).

Refactored `game::explosion`: explosion force is a `float` now, `do_blast` is now included in the `distance_factor` (if >0, there is a blast) and changed function signature to allow invocations with just location and force (as opposed to location, force, fire and shrapnel).
Shrapnel moved to a helper.

Minor balance changes:
- Distance to explosion starts at 0 instead of 1
- Exploding map terrain gets a bit smaller explosion (0.8->0.7)
- EXPLOSIVE_SMALL and FRAG get explosion with as much force as EXPLOSIVE, but with 0.4 distance factor
- NAPALM area is smaller, NAPALM_BIG gets less overall force (it got a bit too big)
- Ammo in fire gets a smaller area (could be changed to just shrapnel now)
- Land mines and booby traps get damage boosts, but area nerf

Overall explosions get more dangerous, but get smaller areas.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
>Land mines and booby traps get damage boosts, but area nerf

Excellent!

**Comment by chaosvolt** [comment]
Ah, nice. That reminds me though, it would be nice if shrapnel range was likewise an adjustable property. As it stands, fragments from a frag round have...what, 2 range I think? Unless the actual range is greater than depicted in tiles mode.

File: src/itype.h [hunk 1]
```diff
@@ -42,9 +42,9 @@ std::string const& default_ammo(std::string const &guntype);
 struct explosion_data {
     // Those 4 values are forwarded to game::explosion.
     int power    = -1;
```
Comment (BevapDin) [inline src/itype.h]: This should probably be a `float`, as it is forwarded to `game::explosion`, which expects a `float`.

File: src/iuse_actor.h [hunk 2]
```diff
@@ -113,9 +113,9 @@ class explosion_iuse : public iuse_actor
         // Those 4 values are forwarded to game::explosion.
         // No explosion is done if power < 0
         int explosion_power;
```
Comment (BevapDin) [inline src/iuse_actor.h]: Probably a `float` as well.

**Pull request merged**